test/e2e/performanceprofile/functests/12_hypershift/hypershift.go

```diff
@@ -0,0 +1,200 @@
+	testutils "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils"
+	testclient "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/client"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/discovery"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/hypershift"
+	testlog "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/log"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/nodepools"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/nodes"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/profiles"
+	hypershiftv1beta1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+	kubeletconfigv1beta1 "k8s.io/kubelet/config/v1beta1"
+)
+
+var _ = Describe("Multiple performance profile in hypershift", Ordered, func() {
+	var profile, secondProfile *performancev2.PerformanceProfile
+	var nodePools []*hypershiftv1beta1.NodePool
+	var np *hypershiftv1beta1.NodePool
+	var workerRTNodes []corev1.Node
+	var err error
+
+	nodeLabel := testutils.NodeSelectorLabels
+	chkKubeletConfig := []string{"cat", "/rootfs/etc/kubernetes/kubelet.conf"}
+	chkKubeletConfigFn := func(ctx context.Context, node *corev1.Node) (string, error) {
+		out, err := nodes.ExecCommand(ctx, node, chkKubeletConfig)
+		if err != nil {
+			return "", err
+		}
+		output := testutils.ToString(out)
+		return output, nil
+	}
+
+	BeforeAll(func() {
+		By("Checking if discovery mode is enabled and performance profile is not found")
+		if discovery.Enabled() && testutils.ProfileNotFound {
+			Skip("Discovery mode enabled, performance profile not found")
+		}
+		workerRTNodes, err = nodes.GetByLabels(nodeLabel)
+		Expect(err).ToNot(HaveOccurred())
+		profile, err = profiles.GetByNodeLabels(nodeLabel)
+		Expect(err).ToNot(HaveOccurred())
+		hostedClusterName, err := hypershift.GetHostedClusterName()
+		Expect(err).ToNot(HaveOccurred())
+		nodePools, err = ListNodePools(context.TODO(), testclient.ControlPlaneClient, hostedClusterName)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(nodePools).ToNot(BeEmpty(), "no node pools found")
+		Expect(len(nodePools)).To(BeNumerically(">=", 2))
+		PrintNodePoolProfiles(nodePools)
+	})
+
+	Context("Multiple Nodepool", Ordered, func() {
+		isolated := performancev2.CPUSet("1-2")
+		reserved := performancev2.CPUSet("0,3")
+		policy := "best-effort"
+
+		It("should verify support for different performance profiles on hosted cluster via multiple node pools", func() {
+			By("Creating a deep copy of the performance profile for the second node pool")
+			secondProfile = profile.DeepCopy()
+			secondProfile.Name = "second-profile"
+			np = nodePools[1]
```

Comment:
how do you make sure the existed profile isn't configured already on nodePools[1]?

Reply:
Ideally, this test suite should be executed after creating a new nodepool, ensuring that no profiles are pre-configured on it. These two test cases are the only ones that deal with the second nodepool. In the AfterAll block, I'm removing the profile, so there shouldn't be any profiles applied left. However, I can add a check before applying the profile to confirm this.

Reply:
Added a check to ensure no profile is configured already with the nodepool, with latest commit.